velox/exec/HashBuild.cpp

```diff
@@ -107,16 +107,18 @@ HashBuild::HashBuild(
         true, // hasProbedFlag
         mappedMemory_);
   } else {
-    // Semi and anti join with no extra filter only needs to know whether there
-    // is a match. Hence, no need to store entries with duplicate keys.
+    // (Left) semi and anti join with no extra filter only needs to know whether
+    // there is a match. Hence, no need to store entries with duplicate keys.
     const bool dropDuplicates = !joinNode->filter() &&
         (joinNode->isSemiJoin() || joinNode->isAntiJoin());
-
+    // Right semi join needs to tag build rows that were probed.
+    const bool needProbedFlag = joinNode->isRightSemiJoin();
+    // Ignore null keys
     table_ = HashTable<true>::createForJoin(
         std::move(keyHashers),
         dependentTypes,
         !dropDuplicates, // allowDuplicates
-        false, // hasProbedFlag
+        needProbedFlag, // hasProbedFlag
```

Comment:
Would it make sense to refactor the encoding if/else block to initialize hasProbedFlag and allowDuplicates flags, than call createForJoin?